Add 32-bit variants for XFERLIST_SRAM_LAYOUT and XFERLIST_EP_INFO

harrisonmutai-arm opened the pull request:
Adds 32-bit versions of the XFERLIST_RW_MEM_LAYOUT and XFERLIST_EP_INFO structures to support booting on 32-bit Arm (aarch32) systems, and corrects the Mbed-TLS heap info heading to fall under the Arm subsection.

sjg20 commented on source/transfer_list.rst:
```diff
@@ -1033,6 +1033,45 @@ passed to later stages for intialisation of Mbed-TLS.
+   * - ep_info
+     - `sizeof(entry_point_info)`
+     - hdr_size
+     - Holds a single `entry_point_info` structure.
```
This structure should be defined here, so please copy over the info from wherever it is kjept.

danh-arm reviewed (changes requested):
Also missing signed-off-by lines

harrisonmutai-arm commented on source/transfer_list.rst:
```diff
@@ -1073,6 +1073,50 @@ subsequent images. It's usage is identical to the 64-bit form represented by
+     - The size of the layout in bytes.
+
+   * - addr
+     - 0x8
```
Sorry that's a typo! It should indeed be a 32-bit address.

danh-arm reviewed (dismissed):
Only minor comments

sjg20 reviewed (changes requested):
I still don't like referring to structures in the guts of a project. You can pull out the C struct definitions and put it in here. That way it can never change without a commit to this project.

harrisonmutai-arm commented:
> I still don't like referring to structures in the guts of a project. You can pull out the C struct definitions and put it in here. That way it can never change without a commit to this project.

Could you clarify what you mean by "pull out the C struct definitions"? I believe I’ve already addressed this in your previous request. I added the 32-bit `entry_point_info_t` C structure [1], and the RW memory layout is already defined in its layout table [2].

[1] https://github.com/FirmwareHandoff/firmware_handoff/pull/54/files#diff-75d1d8518f52351181570a42b6cee154a72b2eb15b3771b969f182b2e537b41eR1171
[2] https://github.com/FirmwareHandoff/firmware_handoff/pull/54/files#diff-75d1d8518f52351181570a42b6cee154a72b2eb15b3771b969f182b2e537b41eR1271
